library/src/main/java/com/novoda/downloadmanager/LiteDownloadMigrationService.java

```diff
@@ -1,97 +1,78 @@
-    private NotificationChannelCreator channelCreator;
     private NotificationManager notificationManager;
-    private MigrationStatus migrationStatus;
 
     @Override
-    public int onStartCommand(Intent intent, int flags, int startId) {
-        super.onStartCommand(intent, flags, startId);
-        Log.d(TAG, "onStartCommand");
-
-        executor.execute(new Runnable() {
-            @Override
-            public void run() {
-                Migrator migrator = MigrationFactory.createVersionOneToVersionTwoMigrator(
-                        getApplicationContext(),
-                        getDatabasePath("downloads.db"),
-                        LiteDownloadMigrationService.this,
-                        channelCreator,
-                        notificationCreator
-                );
-                Log.d(TAG, "Begin Migration: " + migrator.getClass());
-                migrator.migrate();
-            }
-        });
+    public void onCreate() {
+        Log.d(TAG, "onCreate");
+        if (executor == null) {
+            executor = Executors.newSingleThreadExecutor();
+        }
 
-        return START_STICKY;
-    }
+        binder = new MigrationDownloadServiceBinder();
+        notificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
 
-    @Override
-    public void updateNotification(NotificationInformation notificationInformation) {
-        startForeground(notificationInformation.getId(), notificationInformation.getNotification());
+        super.onCreate();
     }
 
     @Override
-    public void stackNotification(NotificationInformation notificationInformation) {
-        stopForeground(true);
-        Notification notification = notificationInformation.getNotification();
-        notificationManager.notify(notificationInformation.getId(), notification);
+    public void setNotificationChannelCreator(NotificationChannelCreator notificationChannelCreator) {
+        this.notificationChannelCreator = notificationChannelCreator;
     }
 
     @Override
-    public void updateMessage(MigrationStatus migrationStatus) {
-        if (migrationCallback != null) {
-            this.migrationStatus = migrationStatus;
-            migrationCallback.onUpdate(migrationStatus);
-        }
+    public void setNotificationCreator(NotificationCreator<MigrationStatus> notificationCreator) {
+        this.notificationCreator = notificationCreator;
     }
 
     @Override
-    public void onCreate() {
-        Log.d(TAG, "onCreate");
-        if (executor == null) {
-            executor = Executors.newSingleThreadExecutor();
-        }
+    public void startMigration(MigrationCallback migrationCallback) {
+        createNotificationChannel();
+        MigrationJob migrationJob = new MigrationJob(getApplicationContext(), getDatabasePath("downloads.db"));
+        migrationJob.addCallback(migrationCallback);
+        migrationJob.addCallback(notificationMigrationCallback);
```

Comment:
`MigrationJob` takes several `callbacks` so that we can easily distinguish what comes from the client and what we do internally. I felt this was better than wrapping them unnecessarily.